src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1036,7 +1040,66 @@ void CodeGen::genSSSE3Intrinsic(GenTreeHWIntrinsic* node)
 //
 void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
 {
-    NYI("Implement SSE41 intrinsic code generation");
+    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    GenTree*       op1         = node->gtGetOp1();
+    GenTree*       op2         = node->gtGetOp2();
+    GenTree*       op3         = nullptr;
+    GenTree*       op4         = nullptr;
+    regNumber      targetReg   = node->gtRegNum;
+    var_types      targetType  = node->TypeGet();
+    var_types      baseType    = node->gtSIMDBaseType;
+
+    regNumber op1Reg = REG_NA;
+    regNumber op2Reg = REG_NA;
+    regNumber op3Reg = REG_NA;
+    regNumber op4Reg = REG_NA;
+    emitter*  emit   = getEmitter();
+
+    if ((op1 != nullptr) && !op1->OperIsList())
+    {
+        op1Reg = op1->gtRegNum;
+        genConsumeOperands(node);
+    }
+
+    switch (intrinsicID)
+    {
+        case NI_SSE41_Blend:
+        case NI_SSE41_DotProduct:
+        {
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, node->gtSIMDBaseType);
+            genHWIntrinsic_FullRangeImm8(node, ins);
```

Comment:
I suggest to wait for https://github.com/dotnet/coreclr/pull/16183 or merge it soon...

Reply:
I'm hoping that I can rebase this PR after #16183 goes in.